### DISCUSSION TITLE
Is it possible to disable the animation feature in GLSP for specific elements?

### DISCUSSION THREAD

**Discussion by rsoika** [eclipse-glsp/glsp#931] [General] [answered]
In GLSP, when I update a GNode (e.g. changing text labels, or resizing) the element shows an nice animation

Nevertheless, I would like to ask whether it is possible to switched off this feature for specific parts of a GNode of for a GNode in general??

![Peek 2023-02-24 17-14](https://user-images.githubusercontent.com/1339124/221229758-672ccd3c-300c-4003-929e-21ad1c9ad39d.gif)

**Comment by tortmayr** [comment, marked as answer]
HI @rsoika,
selective animation is a bit tricky.
First of all in GLSP/sprotty animations are not directly coupled with a model element. Instead they are tied to specific operation/command.

Each server operation results in an [`UpdateModelAction`](https://github.com/eclipse-glsp/glsp/blob/master/PROTOCOL.md#243-updatemodelaction. This action has a flag to indicate whether the update should be animated or not. Currently this animation flag is controlled globally via the [DiagramConfiguration](https://github.com/eclipse-glsp/glsp-server/blob/1bbad9889c1ae31c9e0dbf8abb92152cd84fdda0/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/diagram/DiagramConfiguration.java#L103) and the corresponding actions are dispatched by the [`ModelSubmissionHandler`](https://github.com/eclipse-glsp/glsp-server/blob/1bbad9889c1ae31c9e0dbf8abb92152cd84fdda0/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java#L104).

I admit that this is not ideal and I have created https://github.com/eclipse-glsp/glsp/issues/938 to track this.
In the meantime you would  have to customize the `ModelSubmissionHandler` `OperationActionHandler` in your application to achieve selective animation.

**Comment by rsoika** [comment]
Thanks @tortmayr, 

finally I solved the issue for OpenBPMN by overwriting the `animatedUpdate()` method in my `DiagramConfiguration` 

```java
public class BPMNDiagramConfiguration extends BaseDiagramConfiguration {
......
    @Override
    public boolean animatedUpdate() {
        return false;
    }
....
}
```
